operator/internal/manifests/storage/configure.go

```diff
@@ -236,3 +257,65 @@ func envVarFromValue(name, value string) corev1.EnvVar {
 		Value: value,
 	}
 }
+
+func managedAuthEnabled(opts Options) bool {
+	storeType := opts.SharedStore
+	switch storeType {
+	case lokiv1.ObjectStorageSecretS3:
+		return opts.S3 != nil && opts.S3.STS
+	default:
+		return false
+	}
+}
+
+func setSATokenPath(opts *Options) {
+	var wiToken string
+	switch opts.SharedStore {
+	case lokiv1.ObjectStorageSecretS3:
+		wiToken = saTokenVolumeK8sDirectory
+		if opts.OpenShiftEnabled {
+			wiToken = saTokenVolumeOcpDirectory
+		}
+		opts.S3.WebIdentityTokenFile = wiToken
+	}
+}
+
+func saTokenVolumeMount(opts Options) corev1.VolumeMount {
+	var wiToken string
+	switch opts.SharedStore {
+	case lokiv1.ObjectStorageSecretS3:
+		wiToken = opts.S3.WebIdentityTokenFile
+	}
+	return corev1.VolumeMount{
+		Name:      saTokenVolumeName,
+		MountPath: wiToken,
+	}
+}
+
+func saTokenVolume(opts Options) corev1.Volume {
+	var audience string
+	storeType := opts.SharedStore
+	switch storeType {
+	case lokiv1.ObjectStorageSecretS3:
+		audience = awsDefaultAudience
+		if opts.S3.Audience != "" {
+			audience = opts.S3.Audience
+		}
+	}
```

Comment:
I believe we miss to default the audience for OpenShift ServiceAccount tokens to `openshift` here, right? 
```suggestion
	case lokiv1.ObjectStorageSecretS3:
		audience = awsDefaultAudience
		if opts.S3.Audience != "" {
			audience = opts.S3.Audience
		}
		if opts.OpenShiftEnabled != "" {
			audience = "openshift"
		}
	}
```